kore/src/Kore/Step/Simplification/Builtin.hs

```diff
@@ -42,18 +53,73 @@ simplifyBuiltin
     -> MultiOr (Conditional variable (Builtin (TermLike variable)))
 simplifyBuiltin =
     \case
-        Domain.BuiltinMap _map -> do
-            _map <- sequence _map
-            -- MultiOr propagates \bottom children upward.
-            return (Domain.BuiltinMap <$> sequenceA _map)
+        Domain.BuiltinMap map' -> simplifyInternalMap map'
         Domain.BuiltinList _list -> do
             _list <- sequence _list
             -- MultiOr propagates \bottom children upward.
             return (Domain.BuiltinList <$> sequenceA _list)
-        Domain.BuiltinSet _set -> do
-            _set <- sequence _set
-            return (Domain.BuiltinSet <$> sequenceA _set)
+        Domain.BuiltinSet set' -> simplifyInternalSet set'
         Domain.BuiltinInt int -> (return . pure) (Domain.BuiltinInt int)
         Domain.BuiltinBool bool -> (return . pure) (Domain.BuiltinBool bool)
         Domain.BuiltinString string ->
             (return . pure) (Domain.BuiltinString string)
+
+simplifyInternal
+    ::  ( Ord variable
+        , Show variable
+        , Unparse variable
+        , SortedVariable variable
+        , Traversable t
+        )
+    => (t (TermLike variable) -> Maybe (t (TermLike variable)))
+    -> t (OrPattern variable)
+    -> MultiOr (Conditional variable (t (TermLike variable)))
+simplifyInternal normalizer tOrPattern = do
+    conditional <- Lens.ala Compose traverse tOrPattern
```

Comment:
I changed this to
```.hs
    conditional <- getCompose $ traverse Compose tOrPattern
```
I still like using `Compose` to lift through two `Traversable` at once, instead of an awkward combination of `traverse` and `sequence`.